src/EigenVector.cpp

```diff
@@ -32,6 +34,38 @@ void EigenVector::write_eigen_vector(const std::string &filename,
   }
 }
 
+bool EigenVector::test_trace_sum(const ssize_t t, const bool do_throw) {
+  bool fail = false;
+  Eigen::MatrixXcd VdV = V[t].adjoint() * V[t];
+  const std::complex<double> trace = VdV.trace();
+  const std::complex<double> sum = VdV.sum();
+  // we allow for some deviation per eigenvector, but not much!
+  if( abs( trace.real() - V[t].cols() ) > 2*V[t].cols()*DBL_EPSILON ){
```

Comment:
not sure if this is sufficiently stringent as a test, we should scan around a little and adjust that pre-factor perhaps?

Reply:
Hi, Matthias actually checked the norm of each eigenvector by reading separately with a condition:
if(std::sqrt(norm)!=1 ){
          std::runtime_error("Norm of Eigenvector is incorrect\n");
        }
For 
( abs( trace.real() - V[t].cols() )
I found 2.2737367544e-13
for the cA2.60.32

Reply:
Sure, feel free to add other checks. This is the easiest global check to do. The check that you mention should not be implemented like that however, since you should not (cannot!) test floating point numbers for equality except in situations where you know exactly that there is no rounding.

Actually, could you add a printout of the deviation ```( abs( trace.real() - V[t].cols() )``` to the error message?

So I guess we need to adjust the sensitivity of this to something like

```
4*V[t].cols()*DBL_EPSILON
```

or even higher.